src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -125,9 +123,7 @@ uct_cuda_ipc_mem_add_reg(void *addr, uct_cuda_ipc_memh_t *memh,
             ucs_trace("packed mempool handle and export pointer for %p", addr);
             goto common_path;
         } else {
-            /* VMM, mempool queries failed. Try legacy IPC handle */
-            key->ph.handle_type = UCT_CUDA_IPC_KEY_HANDLE_TYPE_LEGACY;
-            goto legacy_path;
+            goto err;
```

Comment:
maybe if (status!=OK) goto err? to avoid the else

Reply:
@yosefe I'm assuming that this is the suggested change:
```diff
             key->ph.handle_type = UCT_CUDA_IPC_KEY_HANDLE_TYPE_MEMPOOL;
             ucs_trace("packed mempool handle and export pointer for %p", addr);
             goto common_path;
-        } else {
+        }
+
+        if (status != UCS_OK) {
             goto err;

```

This would miss the case where status is UCS_OK and mempool is 0. In this case too, we want to fail because legacy_ipc check, vmm check, and mempool check all failed. For this reason, shouldn't we keep the else case as it is?

Reply:
i think the idea is to check the following on line 106
```
if ((status != UCS_OK) || (mempool == 0)) {
    goto err;
}
```
this way else is not needed and nesting can be reduced